diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/FileStructureRule.kt

```diff
@@ -0,0 +1,161 @@
+import org.cqfn.diktat.common.config.rules.RulesConfig
+import org.cqfn.diktat.ruleset.constants.Warnings.FILE_COLLAPSED_IMPORTS
+import org.cqfn.diktat.ruleset.constants.Warnings.FILE_CONTAINS_ONLY_COMMENTS
+import org.cqfn.diktat.ruleset.constants.Warnings.FILE_INCORRECT_BLOCKS_ORDER
+import org.cqfn.diktat.ruleset.constants.Warnings.FILE_NO_BLANK_LINE_BETWEEN_BLOCKS
+import org.cqfn.diktat.ruleset.constants.Warnings.FILE_UNORDERED_IMPORTS
+import org.cqfn.diktat.ruleset.rules.getDiktatConfigRules
+import org.cqfn.diktat.ruleset.utils.findChildBefore
+import org.cqfn.diktat.ruleset.utils.isChildAfterAnother
+import org.cqfn.diktat.ruleset.utils.isChildBeforeAnother
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.LeafPsiElement
+import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.PsiWhiteSpaceImpl
+import org.jetbrains.kotlin.com.intellij.psi.tree.TokenSet
+import org.jetbrains.kotlin.psi.KtImportDirective
+
+/**
+ * Visitor for checking internal file structure.
+ * 1. Checks file contains not only comments
+ * 2. Ensures the following blocks order: Copyright, Header Kdoc, @file annotation, package name, Import statements,
+ *    top class header and top function header comments, top-level classes or top-level functions
+ * 3. Ensures there is a blank line between these blocks
+ * 4. Ensures imports are ordered alphabetically without blank lines
+ * 5. Ensures there are no collapsed imports
+ */
+class FileStructureRule : Rule("file-structure") {
+
+    private lateinit var configRules: List<RulesConfig>
+    private lateinit var emitWarn: ((offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit)
+    private var isFixMode: Boolean = false
+    private var fileName: String = ""
+
+    override fun visit(node: ASTNode, autoCorrect: Boolean, params: KtLint.Params, emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit) {
+
+        configRules = params.getDiktatConfigRules()
+        isFixMode = autoCorrect
+        emitWarn = emit
+
+        if (node.elementType == ElementType.FILE) {
+            fileName = params.fileName ?: ""
+            if (checkFileHasCode(node)) {
+                checkCodeBlocksOrderAndEmptyLines(node)
+            }
+            checkImportsOrder(node.findChildByType(IMPORT_LIST)!!)
+        }
+    }
+
+    private fun checkFileHasCode(node: ASTNode): Boolean {
+        val codeTokens = TokenSet.andNot(TokenSet.ANY, TokenSet.create(WHITE_SPACE, KDOC, BLOCK_COMMENT, EOL_COMMENT, PACKAGE_DIRECTIVE, IMPORT_LIST))
+        val hasCode = node.getChildren(codeTokens).isNotEmpty()
+        if (!hasCode) {
+            FILE_CONTAINS_ONLY_COMMENTS.warn(configRules, emitWarn, isFixMode, fileName, node.startOffset)
+        }
+        return hasCode
+    }
+
+    private fun checkCodeBlocksOrderAndEmptyLines(node: ASTNode) {
+        val copyrightComment = node.findChildBefore(PACKAGE_DIRECTIVE, BLOCK_COMMENT)
```

Comment:
I have found out that there is my mistake in findChildBefore - I have added empty KDOC to it. Could you please add description?